test/gtest/reducecalculation.hpp

```diff
@@ -274,4 +324,6 @@ struct ReduceCalculationTest : public ::testing::TestWithParam<ReduceCalculation
     int32_t dim;
     miopenReduceCalculationNanPropagation_t nanPropagation;
     miopenReduceCalculationOp_t reduceCalculationOp;
+
+    bool isLogicalCalculation = false;
```

Comment:
simply create a function to check for logicalCalculation type instead of create a separate variable:
```
// Utils
bool IsValidFloatTypes()
{
    return std::is_same<T, float>::value || std::is_same<T, half_float::half>::value ||
           std::is_same<T, bfloat16>::value;
}

bool IsLogicalCalculation()
{
    return reduceCalculationOp == MIOPEN_REDUCE_CALCULATION_ANY ||
           reduceCalculationOp == MIOPEN_REDUCE_CALCULATION_ALL;
}
```